data-gen/review.js

```diff
@@ -0,0 +1,120 @@
+const fs = require('fs');
+const path = require('path');
+const faker = require('faker');
+const { writeToCsv } = require('./utils');
+
+const REVIEWS_PER_GAME = 10;
+let unusedUserIds = new Array(750).fill(0).map((_, idx) => idx + 1);
+
+/**
+ * Generates n unique ids between low and high, inclusive,
+ * with a preference for unused ids
+ * @param {Integer} low
+ * @param {Integer} high
+ * @param {Integer} n
+ * @returns {Array} array length n
+ */
+const generateNUniqueIds = (low, high, n) => {
+  let result = [];
+  for (let i = 0; i < n; i++) {
+    let uniqueId;
+    if (unusedUserIds.length) {
+      let randIdx = Math.floor(Math.random() * unusedUserIds.length);
+      uniqueId = unusedUserIds[randIdx];
+      unusedUserIds.splice(randIdx, 1);
+    } else {
+      do {
+        uniqueId = Math.floor(Math.random() * high) + low;
+      } while (result.includes(uniqueId));
+    }
+    result.push(uniqueId);
+  }
+  return result;
+};
+
+/**
+ * Generates REVIEWS_PER_GAME * 100 units of review data based on seed.sql reviews schema
+ * @returns {Array}: REVIEWS_PER_GAME * 100 elements long
+ */
+const generateReviewData = () => {
+  /**
+   * Rules:
+   * 1. id_user must be between 1-NUM_USERS (750)
+   * 2. id_game must be between 1-100
+   * 3. hours_at_review_time must not exceed hours_on_record
+   * 4. purchase_type must be one of ('direct', 'key')
+   * 5. date_posted must be weight to be more recent
+   * 6. A user may only review a game at most 1 time
+   * 7. Each used reviews at least 1 game
+   */
+
+  // Maps reviewData to id_game, ensuring there are REVIEWS_PER_GAME num of each game id
+  let reviewData = new Array(REVIEWS_PER_GAME * 100).fill(0);
+  reviewData = reviewData.map((_, idx) => (idx - (idx % REVIEWS_PER_GAME)) / REVIEWS_PER_GAME + 1);
+
+  // Generate other data specified in reviews schema
+  // Rules: 1, 2, 6, 7
+  for (let i = 0; i < reviewData.length; i += REVIEWS_PER_GAME) {
+    let gameAndUserIdTuples = generateNUniqueIds(1, 750, REVIEWS_PER_GAME)
+      .map(userId => ({
+        idGame: reviewData[i],
+        idUser: userId
```

Comment:
This is just personal preference, I usually name variables either the same name as the key its being assigned to or via es6 shorthand property names. In this case I guess since idGame isnt assigned via shorthand it makes more sense to manually assign keys to values.

Reply:
Noted & made a commit to change this to be more in line, since it's my preference as well. Overlooked this when I was coding last night